### ISSUE TITLE
Column width calculation issue

### ISSUE THREAD

**Issue by plotn** [dbeaver/dbeaver#19339]
There is a good option - display column description in data grid. Obviously, column descriptions almost always longer than column names and, often too, longer than values. So it looks like:
![image](https://user-images.githubusercontent.com/11477081/224485931-ff37432f-1fe0-4592-8f6c-786c78e07e8e.png)

Could we add an option to include column name/description in column width calculation? Somewhere here:
![image](https://user-images.githubusercontent.com/11477081/224485973-db431b9f-e609-4758-a8de-a24bda19b03c.png)

**Comment by plotn** [comment]
Brief investigation leads me to OPTION_EXCLUDE_COLUMN_NAME_FOR_WIDTH_CALC
But it is always false, as I suggest. 
This is dead or unfinished code? Or I miss something?

**Comment by LonwoLonwo** [comment]
Hello @plotn 

Thanks for the idea.

**Comment by plotn** [comment]
> Hello @plotn
> 
> Thanks for the idea.

Answer my questions please. 
If I right - I can do this myself, but lets approve the implementation details.

**Comment by ShadelessFox** [comment]
Hello @plotn,

This option is affected by the `Calculate column width by cell values` reference.

See `GridLabelProvider#getGridOption`

**Comment by plotn** [comment]
> This option is affected by the `Calculate column width by cell values` reference.

but it is not! is it?

**Comment by ShadelessFox** [comment]
Maybe there's an error in the width calculation logic. Need to investigate.

**Comment by plotn** [comment]
I did.
And nevertheless - what is your opinion - is it suitable to add additional checkbox "Including (in calculation) column name / description" to have both beahaviours?

**Comment by E1izabeth** [comment]
We will discuss it soon - it's controversial. Thank you for asking in advance.

**Comment by ShadelessFox** [comment]
Maybe not a checkbox but a combo box to allow choosing whether you want to automatically compute column widths based on their names, descriptions, both, or none. I think it will suit all possible cases.

**Comment by plotn** [comment]
It was really 'soon' )

**Comment by ShadelessFox** [comment]
We haven't discussed it team-wide yet. It was just my personal opinion.

**Comment by E1izabeth** [comment]
@ShadelessFox I thought about this case too. And maybe we can add case `value` into this combo box options too - otherwise I doubt how combo box will work together with checkbox `Calculate column width by cell values`

**Comment by plotn** [comment]
Why with? Instead

**Comment by E1izabeth** [comment]
Yep, instead of existing checkbox `Calculate column width by cell values`.
And, probably, if checkbox `Show column description in header` is not set, then `calculate column width by` `description` should not work - it seems pointless for me.
But I can't say right now how to achieve it better and if we should do that, because it's not clear what should happen with combo box value, if we turned off checkbox while this setting was set. Maybe there is no need to made things too complicated and I'm just overthinking.

**Comment by plotn** [comment]
what if:
[combobox] Column width calculation: default; values; title and values; title, description (if visible) and values
?

**Comment by E1izabeth** [comment]
Yes, probably it will be ok.

**Comment by plotn** [comment]
> Yes, probably it will be ok.

Look at the implementation please, it became more clear - as for me (see the grid and the option):

![image](https://user-images.githubusercontent.com/11477081/225939224-81ab58fe-6550-41b8-9e49-5f09d3050752.png)

![image](https://user-images.githubusercontent.com/11477081/225939323-1021c9dc-bed2-4b13-92a2-fc74b3cdd274.png)

![image](https://user-images.githubusercontent.com/11477081/225939386-cca5e0d4-e877-49ea-bd7d-f21938016eb5.png)

![image](https://user-images.githubusercontent.com/11477081/225939462-35f945a0-1f44-447d-b13d-b8a6d70fa8fb.png)

Is it ok?

**Comment by ShadelessFox** [comment]
Looks good to me.

**Comment by plotn** [comment]
4th screenshot is broken - grid should be like this:
![image](https://user-images.githubusercontent.com/11477081/225940044-9718ee44-25b9-4213-9829-a95423f02276.png)

(titles and values fit, descrition is not, but visible)

**Comment by plotn** [comment]
> Looks good to me.

then I'll PR this work
